cli/template/base/src/env/server.mjs

```diff
@@ -3,9 +3,21 @@
  * This file is included in `/next.config.mjs` which ensures the app isn't built with invalid env vars.
  * It has to be a `.mjs`-file to be imported there.
  */
-import { serverSchema, serverEnv } from "./schema.mjs";
+import { serverSchema } from "./schema.mjs";
 import { env as clientEnv, formatErrors } from "./client.mjs";
 
+/** @type {{ [key: string]: string | undefined; }} */
+let serverEnv = {};
+
+/**
+ * You can't destruct `process.env` as a regular object, so you have to do
+ * it manually here. This is because Next.js evaluates this at build time,
+ * and only used environment variables are included in the build.
+ */
+Object.keys(serverSchema.shape).forEach(
+  (key) => (serverEnv[key] = process.env[key]),
+);
```

Comment:
```suggestion
/**
 * You can't destruct `process.env` as a regular object, so we do
 * a workaround. This is because Next.js evaluates this at build time,
 * and only used environment variables are included in the build.
 * @type {{ [key: string]: string | undefined; }}
 */
let serverEnv = {};
Object.keys(serverSchema.shape).forEach(
  (key) => (serverEnv[key] = process.env[key]),
);
```